src/utils/plotSpecs/generateEmbeddingContinuousSpec.js

```diff
@@ -153,12 +153,24 @@ const generateSpec = (config) => {
   };
 };
 
-const generateData = (spec, selectedExpression, embeddingData) => {
+const generateData = (spec,
+  selectedExpression,
+  selectedSample,
+  embeddingData,
+  cellSetProperties) => {
+  let filteredData = [];
+  if (selectedSample !== 'All') {
+    const cellIds = Array.from(cellSetProperties[selectedSample].cellIds);
+    filteredData = embeddingData.filter((id) => cellIds.includes(embeddingData.indexOf(id)));
+  } else {
+    filteredData = embeddingData;
+  }
+
```

Comment:
This merges the functions for generating data for continuous embedding. into one function. Basically this part  filters the embedding data for the correct sample, if a sample is selected.